go/enclave/core/batchlimiter.go

```diff
@@ -0,0 +1,80 @@
+package core
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type BatchSizeLimiter struct {
+	Size            uint64
+	ContractAddress common.Address
+	Topic           common.Hash
+}
+
+var ErrInsufficientSpace = errors.New("insufficient space in BatchSizeLimiter")
+
+func NewBatchSizeLimiter(size uint64, contractAddress common.Address, topic common.Hash) *BatchSizeLimiter {
+	return &BatchSizeLimiter{
+		Size:            size,
+		ContractAddress: contractAddress,
+		Topic:           topic,
+	}
+}
+
+func (l *BatchSizeLimiter) AcceptTransaction(tx *types.Transaction) error {
+	if l == nil {
```

Comment:
Are these nil checks needed? Shouldn't the caller have the responsibility of initialising the BSL?